identifier/identifier.go

```diff
@@ -56,3 +105,80 @@ func NewIP(ip netip.Addr) ACMEIdentifier {
 		Value: ip.StringExpanded(),
 	}
 }
+
+// FromCert extracts the Subject Common Name and Subject Alternative Names from
+// a certificate, and returns a slice of ACMEIdentifiers and an error.
+//
+// FromCSR is nearly identical but handles CSRs, and is kept separate so that
+// it's always clear we are handling an untrusted CSR.
+func FromCert(cert *x509.Certificate) ([]ACMEIdentifier, error) {
+	var sans []ACMEIdentifier
+	for _, name := range cert.DNSNames {
+		sans = append(sans, NewDNS(name))
+	}
+	if cert.Subject.CommonName != "" {
+		// Boulder won't generate certificates with a CN that's not also present
+		// in the SANs, but such a certificate is possible. If appended, this is
+		// deduplicated later with Normalize(). We assume the CN is a DNSName,
+		// because CNs are untyped strings without metadata, and we will never
+		// configure a Boulder profile to issue a certificate that contains both
+		// an IP address identifier and a CN.
+		sans = append(sans, NewDNS(cert.Subject.CommonName))
+	}
```

Comment:
Since this function asserts that we're only handling trusted certificates, we can disregard CommonName entirely. We have guarantees elsewhere that anything from the CN is also in the SANs.